src/umpire/resource/NUMAMemoryResourceFactory.cpp

```diff
@@ -0,0 +1,66 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#include "umpire/resource/NUMAMemoryResourceFactory.hpp"
+
+#include "umpire/resource/NUMAMemoryResource.hpp"
+
+#include "umpire/util/Macros.hpp"
+
+#if defined(UMPIRE_ENABLE_NUMA)
+#include <numa.h>
+#endif
+
+namespace umpire {
+namespace resource {
+
+NUMAMemoryResourceFactory::NUMAMemoryResourceFactory(const int numa_node_)
+  : numa_node(numa_node_) {}
+
+std::size_t
+NUMAMemoryResourceFactory::getNodeCount() {
```

Comment:
Can this be something like getNumberOfNumaNodes(), I think that getNodeCount is too ambiguous.